File: spring-data-requery-test/src/test/java/org/springframework/data/requery/config/RequeryTestConfiguration.java
package org.springframework.data.requery.config;

import io.requery.meta.EntityModel;
import io.requery.sql.TableCreationMode;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.requery.configs.AbstractRequeryConfiguration;
import org.springframework.data.requery.domain.Models;
import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
import org.springframework.transaction.annotation.EnableTransactionManagement;

import javax.sql.DataSource;

@Configuration
@EnableTransactionManagement
public class RequeryTestConfiguration extends AbstractRequeryConfiguration {

    @Override
    @Bean
    public EntityModel getEntityModel() {
        return Models.DEFAULT;
    }

    @Override
    public TableCreationMode getTableCreationMode() {
        return TableCreationMode.CREATE_NOT_EXISTS;
    }

    @Bean
    public DataSource dataSource() {
//        return DataSources.ofEmbeddedH2();
        return new EmbeddedDatabaseBuilder()
            .setName("data")
            .setType(EmbeddedDatabaseType.H2)
            .setScriptEncoding("UTF-8")
            .ignoreFailedDrops(true)
            .build();
    }
}

File: spring-data-requery-kotlin/src/test/kotlin/com/coupang/springframework/data/requery/configs/RequeryTestConfiguration.kt
package com.coupang.springframework.data.requery.configs

import com.coupang.kotlinx.data.DataSources
import com.coupang.kotlinx.data.JdbcDrivers
import com.coupang.kotlinx.data.config.DatabaseConfigElement
import com.coupang.kotlinx.data.config.DatabaseSetting
import com.coupang.springframework.data.requery.config.AbstractRequeryConfiguration
import com.coupang.springframework.data.requery.domain.Models
import io.requery.meta.EntityModel
import io.requery.sql.TableCreationMode
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import javax.sql.DataSource

/**
 * RequeryTestConfiguration
 *
 * @author debop@coupang.com
 * @since 18. 5. 30
 */
@Configuration
class RequeryTestConfiguration: AbstractRequeryConfiguration() {

    override fun getEntityModel(): EntityModel = Models.DEFAULT

    override fun getTableCreationMode(): TableCreationMode = TableCreationMode.DROP_CREATE

    @Bean
    fun dataSource(): DataSource {
        val h2 = DatabaseSetting(JdbcDrivers.DRIVER_CLASS_H2,
                                 "jdbc:h2:mem:test;MODE=MYSQL;MVCC=TRUE;",
                                 "sa",
                                 "",
                                 DataSources.MAX_POOL_SIZE,
                                 DataSources.MIN_IDLE_SIZE,
                                 DatabaseConfigElement.TEST_QUERY)

        return DataSources.of(h2)
    }


}